src/main/java/org/jboss/remoting3/EndpointImpl.java

```diff
@@ -422,7 +422,7 @@ public Registration registerService(final String serviceType, final OpenListener
         if (sm != null) {
             sm.checkPermission(RemotingPermission.REGISTER_SERVICE);
         }
-        final RegisteredServiceImpl registeredService = new RegisteredServiceImpl(openListener, optionMap);
+        final RegisteredServiceImpl registeredService = new RegisteredServiceImpl(openListener, EndpointBuilder.merge(optionMap, defaultConnectionOptionMap));
```

Comment:
I'm not so sure about this, although I would have to take a look at the details to explain why i'm not so sure. But from a simplistic point of view, you are merging connection options into an option map for channel creation and the options used for these artifacts differ. There are a lot of OptionMap instances floating around here and they don't all apply to the same creation artifacts.

Reply:
> But from a simplistic point of view, you are merging connection options into an option map for channel creation and the options used for these artifacts differ

Ah, you are correct, I mixed the options used for connection with the ones for channel creation. thanks. Indeed there are many OptionMap instances floating.